crates/sui-graphql-rpc/src/types/intersect.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Merges two filter fields. If both values exist, `merge` is used to combine them, which returns
+/// some combined value if there is some consistent combination, and `None` otherwise. The overall
+/// function returns `Some(None)`, if the filters combined to no filter, `Some(Some(f))` if the
+/// filters combined to `f`, and `None` if the filters couldn't be combined.
+pub(crate) fn field<T>(
+    this: Option<T>,
+    that: Option<T>,
+    merge: impl FnOnce(T, T) -> Option<T>,
+) -> Option<Option<T>> {
+    match (this, that) {
+        (None, None) => Some(None),
+        (Some(this), None) => Some(Some(this)),
+        (None, Some(that)) => Some(Some(that)),
```

Comment:
ah yeah, the issue was I was reviewing this PR very late last night, I see that in cases where one is None and other is Some, we select the Some. clarifies all my qs. Thanks!